docs/guides/schemaless.md

```diff
@@ -0,0 +1,191 @@
+---
+title: Schemaless
+sidebar_position: 4
+---
+
+# Strict schema or schemaless?
+
+Quickwit let's you place the cursor on how strict you would like your schema to be. In other words, it is possible to operate Quickwit with a very strict mapping, in an entirely schemaless manner, and anywhere in between.  Let's see how this works!
+
+## A strict mapping
+
+That's the most straightforward approach.
+As a user, you need to precisely define the list of fields to be ingested by Quickwit.
+
+For instance, a reasonable mapping for an application log could be
+
+```yaml
+version: 0
+
+index_id: "my_app"
+
+doc_mapping:
+  mode: strict # <--- The mode attribute
+  field_mappings:
+    - name: timestamp
+      type: i64
+      fast: true
+    - name: server
+      tokenizer: raw
+    - name: message
+      type: text
+      record: position
+    - name: severity
+      tokenizer: raw
+
+indexing_settings:
+  timestamp_field: timestamp
+
+search_settings:
+  default_search_fields: [severity, message]
+```
+
+The `mode` attribute controls what should be done if an ingested document
+contains a field that is not defined in the document mapping. By default, it takes the `lenient` value. In `lenient` mode, the fields that do not appear in the document mapping will simply be ignored.
+
+If `mode` is set to `strict` on the other hand, documents containing fields
+that are not defined in the mapping will be entirely discarded.
+
+## Schemaless with a partial schema
+
+`mode` can take another value: `dynamic`.
+When set to dynamic, all extra fields will actually be mapped using a catch-all configuration.
+By default, this catch-all configuration index and store all of these fields, but this can be configured by settings the [`dynamic_mapping` attribute](../configuration/index-config#mode).
```

Comment:
```suggestion
By default, this catch-all configuration indexes and stores all of these fields, but this can be configured by settings the [`dynamic_mapping` attribute](../configuration/index-config#mode).
```